Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -19,10 +21,40 @@ namespace BuildXL.Cache.ContentStore.Service.Grpc
     /// An implementation of a CAS copy helper client based on GRPC.
     /// TODO: Consolidate with GrpcClient to deduplicate code. (bug 1365340)
     /// </summary>
-    public class GrpcCopyClient : IShutdown<BoolResult>
+    public sealed class GrpcCopyClient : IShutdown<BoolResult>
     {
+        private static ConcurrentDictionary<(string, int), GrpcCopyClient> _clientPool = new ConcurrentDictionary<(string, int), GrpcCopyClient>();
+
+        /// <summary>
+        /// Use an existing GRPC client if possible, else create a new one.
+        /// </summary>
+        /// <param name="host">Name of the host for the server (e.g. 'localhost').</param>
+        /// <param name="grpcPort">GRPC port on the server.</param>
+        public static GrpcCopyClient Create(string host, int grpcPort)

```

Comment:
nit: could you please move the factory after all the fields (and a private constructor (I suppose we have one).